javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java

```diff
@@ -36,4 +39,12 @@ default VariableDeclarator getVariable(int i) {
     }
 
     N setVariables(NodeList<VariableDeclarator> variables);
+
+    public default Type getCommonType() {
+        return getVariables().stream().map(v -> v.getType())
```

Comment:
Oh hey, the merged getCommonType and yours are not the same. Mine only returns the type of the variables if it is the same for each one. What shall we do about naming?

The thing you are describing has no specific name in the JLS, as far as I know.